src/main/java/httpserver/request/RequestReader.java

```diff
@@ -0,0 +1,29 @@
+package httpserver.request;
+
+import httpserver.interfaces.IReader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class RequestReader implements IReader {
+    private BufferedReader reader;
+
+    public RequestReader(InputStream request) throws IOException {
+        this.reader = new BufferedReader(new InputStreamReader(request));
+    }
+
+    public String readLine() throws IOException {
```

Comment:
I might have missed it but do you have tests for all your new public-facing functions in all the new classes?

Reply:
I do not have tests for RequestReader/ClientHandler. I will add on a separate ticket.